pkg/sfu/buffer/rtpstats.go

```diff
@@ -795,6 +811,32 @@ func (r *RTPStats) GetRtcpSenderReportDataExt() *RTCPSenderReportDataExt {
 	}
 }
 
+func (r *RTPStats) GetExpectedRTPTimestamp(at time.Time) (uint32, error) {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	if !r.initialized {
+		return 0, errors.New("uninitilaized")
+	}
+
+	timeDiff := at.Sub(r.firstTime)
+	rtpDiff := timeDiff.Nanoseconds() * int64(r.params.ClockRate) / 1e9
+	expectedExtRTP := r.extStartTS + uint64(rtpDiff)
+	r.logger.Debugw(
+		"expected RTP timestamp",
+		"firstTime", r.firstTime.String(),
+		"checkAt", at.String(),
+		"timeDiff", timeDiff,
+		"firstRTP", r.extStartTS,
+		"rtpDiff", rtpDiff,
+		"expectedExtRTP", expectedExtRTP,
+		"expectedRTP", uint32(expectedExtRTP),
+		"highestTS", r.highestTS,
+		"highestTime", r.highestTime.String(),
+	)
+	return uint32(expectedExtRTP), nil
```

Comment:
Idea here is to check for time elapsed since first packet was sent and calculate expected time stamp based on clock rate. This is used to decide next time stamp on layer switch.

Reply:
Another idea here (if necessary) is to use something like a PID controller to push/pull the difference between expected and actual when doing sender report so that the timestamp does not drift.